azurelinuxagent/ga/utils/exthandler_utils.py

```diff
@@ -0,0 +1,343 @@
+            self.cpu_limits.append(CpuLimitInstance(cpu_info["cores"], cpu_info["limit_percentage"]))
+            self.cores.append(cpu_info["cores"])
+
+        if DEFAULT_CORES_COUNT not in self.cores:
+            raise ExtensionHandlerConfigurationError("Default CPU limit not set."
+                                              " Core configuration for {0} not present".format(DEFAULT_CORES_COUNT))
+
+        self.cpu_limits = sorted(self.cpu_limits)
+
+    def __str__(self):
+        return self.cpu_limits.__str__()
+
+
+class CpuLimitInstance(object):
+    def __init__(self, cores, limit_percentage):
+        if type(cores) is not int:
+            raise ExtensionHandlerConfigurationError(
+                "Incorrect types for CPU values | field - {0}/ value - {1}/ type - {2}".format("cores", cores,
+                                                                                               type(cores)))
+
+        if type(limit_percentage) is not float and type(limit_percentage) is not int:
+            raise ExtensionHandlerConfigurationError(
+                "Incorrect types for CPU values | field - {0}/ value - {1}/ type - {2}".format("limit_percentage",
+                                                                                               limit_percentage,
+                                                                                               type(limit_percentage)))
+
+        if cores != DEFAULT_CORES_COUNT and cores < 1:
+            raise ExtensionHandlerConfigurationError(
+                "CPU cores value incorrect | field - {0}, {1}/ value - {2}, {3}".format("cores", "limit_percentage",
+                                                                                        cores, limit_percentage))
+
+        if limit_percentage > 100 or limit_percentage < 1:
+            raise ExtensionHandlerConfigurationError(
+                "CPU cores limit_percentage out of range | "
+                "field - {0}, {1}/ value - {2}, {3}".format("cores",
+                                                            "limit_percentage",
+                                                            cores,
+                                                            limit_percentage))
+        self.cores = int(cores)
+        self.limit_percentage = float(limit_percentage)
+
+    def __eq__(self, other):
+        return self.cores == other.cores
+
+    def __lt__(self, other):
+        return self.cores < other.cores
+
+    def __gt__(self, other):
+        return self.cores > other.cores
+
+    def __str__(self):
+        return {"cores": self.cores, "limit_percentage": self.limit_percentage}.__str__()
+
+    def __repr__(self):
+        return {"cores": self.cores, "limit_percentage": self.limit_percentage}.__str__()
+
+
+class MemoryLimits(object):
+    def __init__(self, memory_node):
+        if "max_limit_MBs" in memory_node and "max_limit_percentage" in memory_node:
```

Comment:
we need to be very clear wrt capitalization in our documentation for publishers (or just make comparison case insensitive)